### PR TITLE
Fixed Profile Activity and tab fragments 

### PR DESCRIPTION
**Pull request by shry678**

### CODE REVIEW

File: app/src/main/java/com/example/dine_and_donate/CurrentVouchersFragment.java [hunk 1]
```diff
@@ -13,35 +15,28 @@
...
     private ArrayList<String> mNames = new ArrayList<>();
+    private Context mContext;
 
-    //default constructor
-    public Tab1Fragment() {
-        mImageUrls = new ArrayList<>();
-        mNames = new ArrayList<>();
+    //create view based on data in array lists, inflates the layout of the fragment
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        //super.onCreate(savedInstanceState);
```
Comment (kellymckinnon) [inline app/src/main/java/com/example/dine_and_donate/CurrentVouchersFragment.java]: remove

File: app/src/main/java/com/example/dine_and_donate/LoginActivity.java [hunk 2]
```diff
@@ -24,31 +24,31 @@
 import com.google.firebase.database.FirebaseDatabase;
 import com.google.firebase.database.ValueEventListener;
 
-import java.util.HashMap;
-import java.util.Map;
-
 public class LoginActivity extends AppCompatActivity {
```
Comment (kellymckinnon) [inline app/src/main/java/com/example/dine_and_donate/LoginActivity.java]: can you revert all the changes to this file? it doesn't seem like they are necessary (maybe just a result of merge conflicts)

File: app/src/main/java/com/example/dine_and_donate/MapActivity.java [hunk 3]
```diff
@@ -377,7 +377,7 @@ private void navigationHelper(Class activity) {
 
     private void generateMarkers(String longitude, String latitude) {
         final YelpService yelpService = new YelpService();
```
Comment (kellymckinnon) [inline app/src/main/java/com/example/dine_and_donate/MapActivity.java]: you can probably delete this line now if yelpService isn't used anymore

File: app/src/main/java/com/example/dine_and_donate/ProfileActivity.java [hunk 4]
```diff
@@ -13,118 +13,100 @@
...
     //elements in layout
-    private TabLayout tabLayout;
-    private ViewPagerAdadpter voucherPagerAdadpter;
-    private ViewPager voucherView;
-
-    private TextView userName;
-    private TextView bio;
-    private ImageView profPic;
-    private ImageView blurredPic;
+    private TabLayout mTabLayout;
```
Comment (kellymckinnon) [inline app/src/main/java/com/example/dine_and_donate/ProfileActivity.java]: great!

File: app/src/main/java/com/example/dine_and_donate/ProfileActivity.java [hunk 5]
```diff
@@ -13,118 +13,100 @@
...
+        mLayoutForConsumer = findViewById(R.id.forConsumer);
+        mLayoutForOrg = findViewById(R.id.forOrg);
+        mTabLayout = findViewById(R.id.tabs_profile);
+        mVoucherView = findViewById(R.id.viewpager_id);
+        mVoucherPagerAdapter = new ViewPagerAdapter(getSupportFragmentManager());
+        mVoucherView.setAdapter(mVoucherPagerAdapter);
+        mTabLayout.setupWithViewPager(mVoucherView);
+        mUserName = findViewById(R.id.et_name);
+
+
```
Comment (kellymckinnon) [inline app/src/main/java/com/example/dine_and_donate/ProfileActivity.java]: delete extra line

File: app/src/main/java/com/example/dine_and_donate/ProfileActivity.java [hunk 6]
```diff
@@ -147,4 +129,16 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
             }
         });
     }
+
+    private void navigationHelper(Class activity) {
```
Comment (kellymckinnon) [inline app/src/main/java/com/example/dine_and_donate/ProfileActivity.java]: it seems like this isn't used anywhere? you can delete it if so

File: app/src/main/java/com/example/dine_and_donate/SignUpActivity.java [hunk 7]
```diff
@@ -48,52 +48,54 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                    mPassword.setVisibility(View.VISIBLE);
+                    mSignUpBtn.setVisibility(View.VISIBLE);
+                    mOrgPhone.setVisibility(View.GONE);
+                }  else if(mSpinner.getSelectedItem().toString().equals("Organization")) {
+                    mOrgPhone.setVisibility(View.VISIBLE);
+                    mName.setVisibility(View.VISIBLE);
+                    mEmail.setVisibility(View.VISIBLE);
+                    mSignUpBtn.setVisibility(View.VISIBLE);
+                    mPassword.setVisibility(View.VISIBLE);
                 } else {
```
Comment (kellymckinnon) [inline app/src/main/java/com/example/dine_and_donate/SignUpActivity.java]: is this case possible?

Comment (kellymckinnon) [reply app/src/main/java/com/example/dine_and_donate/SignUpActivity.java]: if not, you only need to change the visibility of mOrgPhone, which would simplify this code a bunch

File: app/src/main/java/com/example/dine_and_donate/StaggeredRecyclerViewAdapter.java [hunk 8]
```diff
@@ -18,71 +19,46 @@
 //bind the data to the view
 public class StaggeredRecyclerViewAdapter extends RecyclerView.Adapter<StaggeredRecyclerViewAdapter.ViewHolder> {
 
+    private ArrayList<String> descriptions = new ArrayList<>();
```
Comment (kellymckinnon) [inline app/src/main/java/com/example/dine_and_donate/StaggeredRecyclerViewAdapter.java]: ah i didn't catch this earlier but let's name these mDescriptions and mImages

File: app/src/main/res/layout/restaurant_info_item.xml [hunk 9]
```diff
@@ -8,15 +8,15 @@
...
+        android:id="@+id/restaurant_iv"
         android:layout_width="82dp"
         android:layout_height="88dp"
         app:srcCompat="@color/common_google_signin_btn_text_dark_focused"
         tools:layout_editor_absoluteX="16dp"
         tools:layout_editor_absoluteY="16dp" />
 
     <TextView
-        android:id="@+id/textView2"
+        android:id="@+id/restaunrant_name_tv"
```
Comment (kellymckinnon) [inline app/src/main/res/layout/restaurant_info_item.xml]: typo: restaurant

**Review by kellymckinnon — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/dine_and_donate/ProfileActivity.java (hunk 6 above)
Comment (shry678) [reply app/src/main/java/com/example/dine_and_donate/ProfileActivity.java]: used in the setUpBottomNav() method

File: app/src/main/java/com/example/dine_and_donate/SignUpActivity.java (hunk 7 above)
Comment (shry678) [reply app/src/main/java/com/example/dine_and_donate/SignUpActivity.java]: if the user goes from the "Consumer" to empty spinner selection, then none of the edit text views should be on the page.

**Pull request closed without merging**